packages/lodestar/src/eth1/jsonRpcHttpClient.ts

```diff
@@ -0,0 +1,98 @@
+// Uses cross-fetch for browser + NodeJS cross compatibility
+// Note: isomorphic-fetch is not well mantained and does not support abort signals
+import fetch from "cross-fetch";
+import {AbortSignal} from "abort-controller";
+import {IJsonRpcClient, IRpcPayload} from "./interface";
+
+interface IRpcResponse<R> {
+  jsonrpc: "2.0";
+  id: number;
+  result?: R;
+  error?: {
+    code: number; // -32601;
+    message: string; // "The method eth_none does not exist/is not available"
+  };
+}
+
+export class JsonRpcHttpClient implements IJsonRpcClient {
+  private url: string;
+
+  constructor(url: string) {
+    this.url = url;
+  }
+
+  /**
+   * Perform RPC request
+   */
+  async fetch<R>({method, params}: IRpcPayload, signal?: AbortSignal): Promise<R> {
+    const res: IRpcResponse<R> = await fetchJson(this.url, {jsonrpc: "2.0", method, params, id: 1}, signal);
+    return parseRpcResponse(res);
+  }
+
+  /**
+   * Perform RPC batched request
+   * Type-wise assumes all requests results have the same type
+   */
+  async fetchBatch<R>(rpcPayloadArr: IRpcPayload[], signal?: AbortSignal): Promise<R[]> {
+    if (rpcPayloadArr.length === 0) return [];
+
+    const resArr: IRpcResponse<R>[] = await fetchJson(
+      this.url,
+      rpcPayloadArr.map(({method, params}, i) => ({jsonrpc: "2.0", method, params, id: i})),
+      signal
+    );
+    return resArr.map(parseRpcResponse);
+  }
+}
+
+function parseRpcResponse<R>(res: IRpcResponse<R>): R {
+  if (res.error) {
+    if (typeof res.error.message === "string") {
+      throw Error(res.error.message);
+    } else if (res.error.code) {
+      throw Error(`JSON RPC error ${res.error.code}`);
+    }
+    throw Error("JSON RPC error");
+  } else if (res.result === undefined) {
+    throw Error("No JSON RPC result");
+  } else {
+    return res.result;
+  }
```

Comment:
It might be better to add rpc request payload to error messages, otherwise we will be clueless about what failed